tests/Unit/RuleSet/RuleSetTest.php

```diff
@@ -1165,6 +1165,84 @@ public function getRulesAssocKeysRulesByPropertyName(): void
         }
     }
 
+    /**
+     * @test
+     *
+     * @param list<string> $propertyNamesToSet
+     * @param list<string> $matchingPropertyNames
+     *
+     * @dataProvider providePropertyNamesAndSearchPatternAndMatchingPropertyNames
+     */
+    public function getRulesAssocWithPatternReturnsAllMatchingPropertyNames(
+        array $propertyNamesToSet,
+        string $searchPattern,
+        array $matchingPropertyNames
+    ): void {
+        $this->setRulesFromPropertyNames($propertyNamesToSet);
+
+        $result = $this->subject->getRulesAssoc($searchPattern);
+
+        foreach ($matchingPropertyNames as $expectedMatchingPropertyName) {
+            self::assertContains($expectedMatchingPropertyName, \array_keys($result));
```

Comment:
If I understand this code correctly, we can use `assertArrayHasKey` here.

https://docs.phpunit.de/en/9.6/assertions.html#assertarrayhaskey

Reply:
Yes.  I didn't think of that.